spock-core/src/main/java/spock/lang/Snapshotter.java

```diff
@@ -0,0 +1,295 @@
+    public String unwrap(String string) {
+      return string;
+    }
+  }
+
+  public static final class Store {
+    private final IterationInfo iterationInfo;
+    private final boolean updateSnapshots;
+    private final String extension;
+    private final Charset charset;
+    private final Path specPath;
+
+    public Store(IterationInfo iterationInfo, Path rootPath, boolean updateSnapshots, String extension, Charset charset) {
+      this.iterationInfo = iterationInfo;
+      this.updateSnapshots = updateSnapshots;
+      this.extension = extension;
+      this.charset = charset;
+
+      Class<?> specClass = iterationInfo.getFeature().getSpec().getReflection();
+      specPath = rootPath
+        .resolve(specClass.getPackage().getName().replace('.', '/'))
+        .resolve(specClass.getSimpleName().replace('$', '/')); // use subdirectories for inner classes
+    }
+
+    private static String calculateSafeUniqueName(String extension, IterationInfo iterationInfo, String snapshotId) {
+      Checks.checkArgument(snapshotId.length() <= 100,
+        () -> String.format("'snapshotId' is too long, only 100 characters are allowed, but was %d: %s", snapshotId.length(), snapshotId));
+
+      FeatureInfo feature = iterationInfo.getFeature();
+      String safeName = sanitize(feature.getName());
+      String featureId = feature.getFeatureMethod().getReflection().getName().substring("$spock_feature_".length());
+      String iterationIndex = feature.isParameterized() ? String.format("-[%d]", iterationInfo.getIterationIndex()) : "";
+      String snapshotIdSuffix = snapshotId.isEmpty() ? "" : "-" + sanitize(snapshotId);
+
+      int uniqueSuffixLength = 1 + featureId.length() + 1 + extension.length() + iterationIndex.length() + snapshotIdSuffix.length();
+      if (safeName.length() + uniqueSuffixLength > 250) {
+        safeName = safeName.substring(0, 250 - uniqueSuffixLength);
+        return String.format("%s%s-%s%s.%s", safeName, snapshotIdSuffix, featureId, iterationIndex, extension);
+      }
+      return String.format("%s%s%s.%s", safeName, snapshotIdSuffix, iterationIndex, extension);
+    }
+
+    @NotNull
+    private static String sanitize(String snapshotId) {
+      return snapshotId.replaceAll("[^a-zA-Z0-9]", "_");
+    }
+
+    public Optional<String> loadSnapshot(String snapshotId) {
+      Path snapshotPath = specPath.resolve(calculateSafeUniqueName(extension, iterationInfo, snapshotId));
+      if (Files.exists(snapshotPath)) {
+        try {
+          return Optional.of(IoUtil.getText(snapshotPath, charset));
+        } catch (IOException e) {
+          throw new UncheckedIOException("Failure while trying to load Snapshot: " + snapshotPath, e);
+        }
+      }
+      return Optional.empty();
+    }
+
+
+    public void saveSnapshot(String snapshotId, String value) {
+      Path snapshotPath = specPath.resolve(calculateSafeUniqueName(extension, iterationInfo, snapshotId));
+      specPath.toFile().mkdirs();
```

Comment:
Why not use `java.nio.file.Files.createDirectories()`, which throws an exception with a bit of better error message, if that is not possible?